src/Elastic.Apm/Helpers/StacktraceHelper.cs

```diff
@@ -32,8 +32,8 @@ internal static List<Stacktrace> GenerateApmStackTrace(StackFrame[] frames, Abst
 			}
 			catch (Exception e)
 			{
-				logger.LogWarning($"Failed capturing stacktrace for {capturingFor}");
-				logger.LogDebug($"{e.GetType().Name}: {e.Message}");
+				logger?.LogWarning(e, nameof(StacktraceHelper), "Failed capturing stacktrace for {ApmContext}", capturingFor);
```

Comment:
IIUC "nameof(StacktraceHelper)" argument will be used as "message" parameter by LogWarning.